maven/rules.bzl

```diff
@@ -301,6 +327,7 @@ assemble_maven = rule(
 
 MavenDeploymentInfo = provider(
     fields = {
+        'packaging': 'The type of target to publish (jar, war, aar, etc.)',
```

Comment:
Given that we are deploying either a JAR or an AAR file, I think it would be more intuitive if we simply read the extension of the file being passed into the `deploy_maven` rule (instead of relying on a new field in `MavenDeploymentInfo`) to determine the `packaging` used in `deploy.py`.